src/WebJobs.Script.WebHost/Management/IInstanceManager.cs

```diff
@@ -13,7 +13,7 @@ public interface IInstanceManager
 
         Task<string> ValidateContext(HostAssignmentContext assignmentContext);
 
-        bool StartAssignment(HostAssignmentContext assignmentContext);
+        Task<bool> StartAssignment(HostAssignmentContext assignmentContext);
```

Comment:
Maybe we should rename this. It is no longer a `Start` anymore with the `Task`. Maybe `AssignHostAsync`?